main/common/aperture.js

```diff
@@ -18,34 +16,41 @@ const {hasMicrophoneAccess} = require('./system-permissions');
 
 const settings = require('./settings');
 const {track} = require('./analytics');
+const plugins = require('./plugins');
 
 const aperture = createAperture();
 const {audioDevices, videoCodecs} = createAperture;
 
 // eslint-disable-next-line no-unused-vars
 const recordHevc = videoCodecs.has('hevc');
 
-let wasDoNotDisturbAlreadyEnabled;
 let lastUsedSettings;
+let recordingPlugins = [];
+const serviceState = new Map();
+let apertureOpts;
 
 let past;
 
-const cleanup = () => {
-  const {
-    hideDesktopIcons,
-    doNotDisturb
-  } = lastUsedSettings;
+const callPlugins = async method => Promise.all(recordingPlugins.map(async ({plugin, service}) => {
+  if (service[method] && typeof service[method] === 'function') {
+    try {
+      await service[method]({
+        options: apertureOpts,
+        state: serviceState.get(service.title),
+        config: plugin.config
```

Comment:
This will be replaced with a context similar to the share service, they will share a bunch of common fields, so I'll do an abstraction